File: experiments/006-SmalltalkVsJava/cells/java-gradle_1-Empty_free/2022-1c-recuperatorio-parcial1/20260915-110924-39025/analysis/own-tests-project/src/main/java/customerimporter/NewZipCode.java
package customerimporter;

import java.util.function.Supplier;

public class NewZipCode extends ZipCode {

    // instance creation

    static boolean appliesTo(String aZipCode) {
        return Character.isLetter(aZipCode.charAt(0));
    }

    // error messages

    public static String invalidNewZipCodeErrorDescription() {
        return "Invalid new zipcode";
    }

    // initialization

    NewZipCode(String aCode) {
        super(aCode);
    }

    // zip code

    public Object value() {
        return code();
    }

    // type

    public boolean isNew() {
        return true;
    }

    public Object newZipCodeIfNone(Supplier<Object> aNoneBlock) {
        return value();
    }

    // validation

    protected void assertIsValid() {
        // "B1636BBE" size 8
        assertThat(code().length() == 8);
        assertThat(Characters.areAllDigits(oldZipCodePart()));
        assertThat(Integer.parseInt(oldZipCodePart()) > 999);
        assertThat(Characters.areAllLetters(code().substring(code().length() - 3)));
    }

    protected String invalidZipCodeErrorDescription() {
        return invalidNewZipCodeErrorDescription();
    }

    // validation - private

    private String oldZipCodePart() {
        return code().substring(1, 5);
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-213435-24827/analysis/own-tests-project/src/customerimporter/new_zip_code.py
from customerimporter.zip_code import ZipCode


class NewZipCode(ZipCode):

    # instance creation

    @classmethod
    def can_represent(cls, a_zip_code):
        return a_zip_code[0].isalpha()

    # error messages

    @classmethod
    def invalid_value_error_description(cls):
        return "Invalid new zipcode"

    # validation

    def assert_is_valid(self):
        # "B1636BBE" size 8
        self._assert_that(len(self._value) == 8)
        self._assert_that(self._are_all_digits(self._old_zip_code_part()))
        self._assert_that(int(self._old_zip_code_part()) > 999)
        self._assert_that(self._are_all_letters(self._value[-3:]))

    # validation - private

    def _old_zip_code_part(self):
        return self._value[1:5]

    # zip code

    def code(self):
        return self._value

    # new zip code

    def has_new_zip_code(self):
        return True

    def new_zip_code_if_none(self, a_none_block):
        return self.code()
